dbt_airflow_factory/builder.py

```diff
@@ -73,52 +105,110 @@ def _create_task_group_for_model(
+                    filter(self._is_model_run_task, node["depends_on"]["nodes"])
                 )
-        return tasks
+                if len(model_dependencies) <= 1:
+                    continue
+                model_dependencies.sort()
+                if tuple(model_dependencies) not in tests_with_more_deps:
+                    tests_with_more_deps[tuple(model_dependencies)] = []
+                tests_with_more_deps[tuple(model_dependencies)].append(node["name"])
+
+        multiple_deps_test_tasks = {}
+        for depends_on_tuple, test_names in tests_with_more_deps.items():
+            multiple_deps_test_tasks[depends_on_tuple] = ModelExecutionTask(
+                self._make_dbt_multiple_deps_test_task(test_names, depends_on_tuple),
+                None,
+            )
+
+        return tasks, multiple_deps_test_tasks
 
     def _create_tasks_dependencies(
-        self, manifest: dict, tasks: Dict[str, ModelExecutionTask]
+        self,
+        manifest: dict,
+        tasks: Dict[str, ModelExecutionTask],
+        multiple_deps_test_tasks: Dict[tuple, ModelExecutionTask],
     ) -> ModelExecutionTasks:
-        starting_tasks = list(tasks.keys())
-        ending_tasks = list(tasks.keys())
+        result_tasks = dict(tasks)
+        for task_test_tuple, task_test in multiple_deps_test_tasks.items():
+            result_tasks[self._build_multiple_deps_test_name(task_test_tuple)] = task_test
+        starting_tasks = list(result_tasks.keys())
+        ending_tasks = list(result_tasks.keys())
+
         for node_name in tasks.keys():
             for upstream_node in manifest["nodes"][node_name]["depends_on"]["nodes"]:
                 if self._is_model_run_task(upstream_node):
                     # noinspection PyStatementEffect
-                    (tasks[upstream_node].get_end_task() >> tasks[node_name].get_start_task())
+                    (
+                        result_tasks[upstream_node].get_end_task()
+                        >> result_tasks[node_name].get_start_task()
+                    )
                     if node_name in starting_tasks:
                         starting_tasks.remove(node_name)
                     if upstream_node in ending_tasks:
                         ending_tasks.remove(upstream_node)
-        return ModelExecutionTasks(tasks, starting_tasks, ending_tasks)
+
+        for depends_on_tuple in multiple_deps_test_tasks.keys():
+            node_name = self._build_multiple_deps_test_name(depends_on_tuple)
+            for upstream_node in depends_on_tuple:
+                # noinspection PyStatementEffect
+                (
+                    result_tasks[upstream_node].get_end_task()
+                    >> result_tasks[node_name].get_start_task()
+                )
+                if node_name in starting_tasks:
+                    starting_tasks.remove(node_name)
+                if upstream_node in ending_tasks:
+                    ending_tasks.remove(upstream_node)
```

Comment:
Looks a little like code duplication, maybe some common part would be extracted?